src/Aspire.Hosting/ApplicationModel/ResourceCommandService.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A service to execute resource commands.
+/// </summary>
+public class ResourceCommandService : IDisposable
+{
+    private readonly ConcurrentDictionary<string, IResource> _resources;
+    private readonly CancellationTokenSource _cts;
+    private readonly ResourceLoggerService _resourceLoggerService;
+    private readonly IServiceProvider _serviceProvider;
+
+    internal ResourceCommandService(ResourceNotificationService resourceNotificationService, ResourceLoggerService resourceLoggerService, IServiceProvider serviceProvider)
+    {
+        _resourceLoggerService = resourceLoggerService;
+        _serviceProvider = serviceProvider;
+        _cts = new();
+        _resources = new ConcurrentDictionary<string, IResource>(StringComparers.ResourceName);
+
+        var cancellationToken = _cts.Token;
+
+        Task.Run(async () =>
+        {
+            await foreach (var @event in resourceNotificationService.WatchAsync().WithCancellation(cancellationToken).ConfigureAwait(false))
```

Comment:
I think we should add a new internal api to go from resource id to resource here. I don't know if we need the ID anywhere else but the dashboard?